pkg/metrics/registry.go

```diff
@@ -27,4 +32,22 @@ type RegistererGatherer interface {
 
 // Registry is a prometheus registry for storing metrics within the
 // controller-runtime.
-var Registry RegistererGatherer = prometheus.NewRegistry()
+var Registry RegistererGatherer = NewRegistry()
+
+func NewRegistry() *prometheus.Registry {
+	r := prometheus.NewRegistry()
+
+	// default registers
+	r.MustRegister(
+		collectors.NewGoCollector(
+			collectors.WithGoCollectorRuntimeMetrics(
+				collectors.MetricsGC,
+				collectors.MetricsScheduler,
+				collectors.MetricsMemory,
+				collectors.GoRuntimeMetricsRule{Matcher: regexp.MustCompile(`^/sync/.*`)},
+			),
+		),
+		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
```

Comment:
https://github.com/kubernetes-sigs/controller-runtime/blob/1ec7c1b76e98efd4f177eb8cd96050596b0fa3d3/pkg/internal/controller/metrics/metrics.go#L90

We do this here, do we need to do this here as well?

Reply:
Thanks @troy0820 
Move to kg/internal/controller/metrics better. 
Fixed.